napari/utils/_magicgui.py

```diff
@@ -177,9 +177,9 @@ def add_layer_data_tuples_to_viewer(gui, result, return_type):
             # then try to update the viewer layer with that name.
             try:
                 layer = viewer.layers[layer_datum[1].get('name')]
-                layer.data = layer_datum[0]
                 for k, v in layer_datum[1].items():
                     setattr(layer, k, v)
+                layer.data = layer_datum[0]
```

Comment:
Do we still need to swap the order here?

Reply:
That's in swapped form now - set the attributes, and then using those new attributes we set the `data` - allows us to first set the `shape_type` as `polygon`, for example, and then pass the vertices through - if the data contains a `shape_type`, that overrides whatever metadata we may have set